src/components/editor/SinopiaResourceTemplates.jsx

```diff
@@ -23,63 +23,69 @@ class SinopiaResourceTemplates extends Component {
   }
 
   async componentDidUpdate(prevProps) {
-    if (this.props.updateKey > prevProps.updateKey) {
-      const groupPromise = getGroups().then(data => {
-        return data
-      }).catch(() => {})
+    if (this.props.updateKey <= prevProps.updateKey)
+      return
 
-      await this.fulfillGroupPromise(groupPromise).then(async () => {
-        this.state.groupData.map(group => {
+    getGroups()
+      .then(data => {
+        const contains = [].concat(data.response.body.contains)
+        this.setState({groupData: contains})
+      })
+      .catch(error => {
+        this.setState({message: error})
+      })
+      .then(() => {
+        return this.state.groupData.map(group => {
           const groupName = this.resourceToName(group)
-          listResourcesInGroupContainer(groupName).then((data) => {
-            this.fulfillGroupData(data)
-          }).catch(() => {})
+          return listResourcesInGroupContainer(groupName)
+        })
+      })
+      .then(groupResources => {
+        Promise.all(groupResources).then(responses => {
+          responses.forEach(group => {
+            this.handleGroupResponse(group)
+          })
         })
       })
-    }
   }
 
-  fulfillGroupPromise = (promise) => {
-    promise.then(data => {
-      const contains = [].concat(data.response.body.contains)
-      this.setState({groupData: contains})
-    }).catch((error) => {
-      this.setState({message: error})
-    })
-    return promise
-  }
+  handleGroupResponse = data => {
+    if (data.response.body === null)
+      return
 
-  fulfillGroupData = (data) => {
     const groupName = this.resourceToName(data.response.body['@id'])
+    // Array-ify the value of `contains` and filter out falsy values like null and undefined
+    const contains = [].concat(data.response.body.contains).filter(Boolean)
+
+    contains.forEach(child => {
+      const name = this.resourceToName(child)
+
+      getResourceTemplate(name, groupName).then(data => {
+        const rt = data.response.body
+        this.setState({
+          tempState: {
+            key: rt.id,
+            name: rt.resourceLabel,
+            uri: child,
+            id: rt.id,
+            group: groupName
+          }
+        })
+        const joined = this.state.templatesForGroup.slice(0)
```

Comment:
Jeremy: a bit uncomfy with `slice`.   Suggests `assigned` -- semantically clearer.  Also, with slice, a blank or empty object, it will error.

Reply:
`Object.assign()`. Though the spread operator (`...`) is nice too.